src/__tests__/createSchemaForm.test.tsx

```diff
@@ -1847,4 +1855,84 @@ describe("createSchemaForm", () => {
     const inputs = screen.getAllByTestId(/dynamic-array-input/);
     expect(inputs.length).toBe(3);
   });
+  describe("CustomChildComponent", () => {
+    it("should not drop focus on rerender", async () => {
+      const schema = z.object({
+        fieldOne: z.string().regex(/moo/),
+        fieldTwo: z.string(),
+      });
+
+      const Form = createTsForm([[z.string(), TextField]] as const, {
+        FormComponent: ({
+          children,
+        }: {
+          onSubmit: () => void;
+          children: ReactNode;
+        }) => {
+          const { isSubmitting } = useFormState();
+          return (
+            <form>
+              {children}
+              {isSubmitting}
+            </form>
+          );
+        },
+      });
+
+      const TestComponent = () => {
+        const form = useForm<z.infer<typeof schema>>({
+          mode: "onChange",
+          resolver: zodResolver(schema),
+        });
+        const values = {
+          ...form.getValues(),
+          ...useWatch({ control: form.control }),
+        };
+
+        return (
+          <Form
+            form={form}
+            schema={schema}
+            defaultValues={{}}
+            props={{
+              fieldOne: {
+                testId: "fieldOne",
+                beforeElement: <>Moo{JSON.stringify(values)}</>,
+              },
+              fieldTwo: { testId: "fieldTwo" },
+            }}
+            onSubmit={() => {}}
+          >
+            {(fields) => {
+              const { isDirty } = useFormState();
```

Comment:
i added this to the test and made it work, but maybe the assumption should be that it doesn't? I think hooks strictly speaking are not completely safe to use in a render prop..